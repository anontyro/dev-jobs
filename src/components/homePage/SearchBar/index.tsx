import React, { useState } from "react";
import styled from "@emotion/styled";
import { getJobList } from "../../../utils/getJsonData";
import CompanyData from "../../../interfaces/companyData";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch, faMapMarker } from "@fortawesome/free-solid-svg-icons";
import InitalJobQuery from "../../../interfaces/search/initalJobQuery";
import { filterCompanyList } from "../../../utils/search/filterCompanys";

const SearchBarContainer = styled.div`
  margin-top: -2.5rem;
  margin-bottom: 5rem;
  .card-content.search {
    display: flex;
    width: 100%;
    justify-content: space-evenly;
    align-items: center;
    .search-query {
      width: 100%;
    }
    .search-location {
      width: 100%;
    }
    .search-controls {
      width: 100%;
      display: flex;
      justify-content: flex-end;
      label {
        display: flex;
        align-items: center;
      }
      button {
        .search-icon {
          display: none;
        }
      }
    }
  }
  @media (max-width: 768px) {
    margin-bottom: 3rem;
    .card-content.search {
      .search-query {
      }
      .search-location {
        display: none;
      }
    }
  }
  @media (max-width: 425px) {
    margin-top: -1rem;
    .card-content.search {
      .search-query {
        .search-icon {
          display: none;
        }
      }
      .search-controls {
        label {
          display: none;
        }
        button {
          min-width: 1rem;
          span.search-text {
            display: none;
          }
          .search-icon {
            display: inline-block;
            margin: auto;
          }
        }
      }
    }
  }
`;

const initalJobQuery: InitalJobQuery = {
  title: "",
  location: "",
  isFullTime: false,
};

interface SearachBarProps {
  jobList: CompanyData[];
  setJobList: (val: CompanyData[]) => void;
}

const SearchBar: React.FC<SearachBarProps> = ({ jobList, setJobList }) => {
  const [jobQuery, setJobQuery] = useState<InitalJobQuery>({
    ...initalJobQuery,
  });

  const searchJobs = () => {
    const companyList = getJobList();
    const results = filterCompanyList(jobQuery, ...companyList);
    setJobList(results);
  };

  return (
    <SearchBarContainer className="card">
      <div className="card-content search">
        <div className="search-query">
          <FontAwesomeIcon className="search-icon" icon={faSearch} />
          <input
            className="input-full"
            type="text"
            placeholder="Filter by title..."
            value={jobQuery.title}
            onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
              setJobQuery({ ...jobQuery, title: e.target.value })
            }
          />
        </div>
        <div className="search-location">
          <FontAwesomeIcon className="search-icon" icon={faMapMarker} />
          <input
            className="input-full"
            type="text"
            placeholder="Filter by location..."
            value={jobQuery.location}
            onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
              setJobQuery({ ...jobQuery, location: e.target.value })
            }
          />
        </div>
        <div className="search-controls">
          <label htmlFor="">
            <span>Full Time Only</span>
            <input
              type="checkbox"
              name="full-time-only"
              id="full-time"
              checked={jobQuery.isFullTime}
              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                setJobQuery({ ...jobQuery, isFullTime: e.target.checked })
              }
            />
          </label>
          <button className="button default" onClick={searchJobs}>
            <span className="search-text">Search</span>
            <FontAwesomeIcon className="search-icon" icon={faSearch} />
          </button>
        </div>
      </div>
    </SearchBarContainer>
  );
};

export default SearchBar;
