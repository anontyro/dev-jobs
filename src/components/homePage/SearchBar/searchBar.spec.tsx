import React from "react";
import renderer from "react-test-renderer";
import SearchBar from ".";

describe("SearchBar tests", () => {
  it("should render without crashing", () => {
    const tree = renderer
      .create(<SearchBar jobList={[]} setJobList={(val: any) => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
