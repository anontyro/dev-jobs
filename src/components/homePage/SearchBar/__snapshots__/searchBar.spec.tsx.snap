// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar tests should render without crashing 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-9nslzq{margin-top:-2.5rem;margin-bottom:5rem;}.css-9nslzq .card-content.search{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;-webkit-box-pack:space-evenly;-ms-flex-pack:space-evenly;-webkit-justify-content:space-evenly;justify-content:space-evenly;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.css-9nslzq .card-content.search .search-query{width:100%;}.css-9nslzq .card-content.search .search-location{width:100%;}.css-9nslzq .card-content.search .search-controls{width:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:end;-ms-flex-pack:end;-webkit-justify-content:flex-end;justify-content:flex-end;}.css-9nslzq .card-content.search .search-controls label{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.css-9nslzq .card-content.search .search-controls button .search-icon{display:none;}@media (max-width: 768px){.css-9nslzq{margin-bottom:3rem;}.css-9nslzq .card-content.search .search-location{display:none;}}@media (max-width: 425px){.css-9nslzq{margin-top:-1rem;}.css-9nslzq .card-content.search .search-query .search-icon{display:none;}.css-9nslzq .card-content.search .search-controls label{display:none;}.css-9nslzq .card-content.search .search-controls button{min-width:1rem;}.css-9nslzq .card-content.search .search-controls button span.search-text{display:none;}.css-9nslzq .card-content.search .search-controls button .search-icon{display:inline-block;margin:auto;}}",
      }
    }
    data-emotion="css 9nslzq"
  />,
  <div
    className="card css-9nslzq"
  >
    <div
      className="card-content search"
    >
      <div
        className="search-query"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-search fa-w-16 search-icon"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <input
          className="input-full"
          onChange={[Function]}
          placeholder="Filter by title..."
          type="text"
          value=""
        />
      </div>
      <div
        className="search-location"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-map-marker fa-w-12 search-icon"
          data-icon="map-marker"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <input
          className="input-full"
          onChange={[Function]}
          placeholder="Filter by location..."
          type="text"
          value=""
        />
      </div>
      <div
        className="search-controls"
      >
        <label
          htmlFor=""
        >
          <span>
            Full Time Only
          </span>
          <input
            checked={false}
            id="full-time"
            name="full-time-only"
            onChange={[Function]}
            type="checkbox"
          />
        </label>
        <button
          className="button default"
          onClick={[Function]}
        >
          <span
            className="search-text"
          >
            Search
          </span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search fa-w-16 search-icon"
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
]
`;
